sequence/src/main/java/com/vitreoussoftware/bioinformatics/sequence/io/reader/BufferFileStreamReader.java

```diff
@@ -0,0 +1,228 @@
+        private int bufferSize = DEFAULT_BUFFER_SIZE;
+    }
+
+
+
+    /**
+     * Returns if the End of File has been reached, indicating no more data can be read
+     *
+     * @return true if End of File reached, false otherwise
+     */
+    public boolean isEof() {
+        if (length == -1) // We've reached EoF
+            return true;
+        if (index < length) // We haven't reached EoF
+            return false;
+
+        // We don't know yet
+        return bufferData() == -1;
+    }
+
+    /**
+     * Does the current position satisfy the predicate?
+     *
+     * @param predicate The {@link Predicate} to test
+     *
+     * @return true if the predicate matches, false if it does not OR eof {@see isEof}
+     */
+    public boolean is(@NonNull final Predicate<Character> predicate) {
+        if (index >= length)
+            bufferData();
+        return index < length && predicate.test(buffer[index]);
+    }
+
+    /**
+     * Does the current position match the {@see character}
+     * @param character the character to compare with
+     *
+     * @return true if they match, false if they do not OR eof {@see isEof}
+     */
+    public boolean is(char character) {
+        if (index >= length)
+            bufferData();
+        return index < length && character == buffer[index];
+    }
+
+    /**
+     * Ignore characters in the buffer until the {@link String} {@see flag} is found
+     *
+     * @param flag The {@link String} to search for
+     */
+    public void dropUntil(String flag) {
+        final int maxPosition = flag.length();
+        int position = 0;
+
+        while (position != maxPosition) {
+            if (index >= length)
+                bufferData();
+            if (length == -1)
+                return;
+            if (buffer[index] == flag.charAt(position)) {
+                position++;
+            }
+            else {
+                position = 0;
+            }
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer until the {@link Predicate} {@see predicate} evaluates to true
+     * @param predicate The {@link Predicate} to test
+     */
+    public void dropUntil(Predicate<Character> predicate) {
+        while (true) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || predicate.test(buffer[index]))
+                return;
+
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer while they are whitespace
+     */
+    public void dropWhileWhitespace() {
+        while (true) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || !Character.isWhitespace(buffer[index]))
+                return;
+
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer until they are whitespace
+     */
+    public void dropUntilWhiteSpace() {
+        while (true) {
```

Comment:
Yep. I do find it a little hard to differentiate dropUntil(x -> !Character.isWhitespace(x)) and dropUntil(x-> Character.isWhitespace(x)) when scanning the code as well. You can get around this using dropUntil(Character::isWhitespace) instead to differentiate it from the ! case. 

I found that dropping whitespace, taking until whitespace, or dropping until whitespace were just very common actions so moving them into full functions makes it a bit easier and lambdas have a performance hit over direct reference.